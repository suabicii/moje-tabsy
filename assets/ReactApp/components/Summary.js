import React from "react";
import DrugList from "./DrugList";
import dayjs from "dayjs";
import StockStatusChecker from "./StockStatusChecker";
import {useSelector} from "react-redux";
import {sortedDrugsSelector} from "../features/drugs/drugsSlice";
import DosingMoments from "./DosingMoments";

function Summary() {
    const drugList = useSelector(sortedDrugsSelector);

    return (
        <>
            <h1 className="text-center mt-5 mt-md-0">Podsumowanie</h1>
            <DrugList isEditMode={false}/>
            {/* SCHEDULE AND STOCK STATUS */}
            <div className="row mt-3">
                <div className="col-md-6">
                    <div className="card">
                        <div className="card-header text-center">
                            Dzisiaj muszę zażyć <i className="fa-solid fa-clock"></i>
                        </div>
                        <div className="card-body">
                            <ul className="list-unstyled">
                                {drugList.map(({dosing, dosingMoments, id, name, unit}) =>
                                    (
                                        <div key={`${name}${id}`}>
                                            <strong
                                                data-testid={`schedule-drugName${id}`}>{name} </strong> – {dosing} {unit}:
                                            <DosingMoments content={dosingMoments} drugId={id}/>
                                        </div>
                                    ))
                                }
                            </ul>
                        </div>
                    </div>
                </div>
                <div className="col-md-6 mt-3 mt-lg-0">
                    <div className="card">
                        <div className="card-header text-center">
                            Stan zapasów <i className="fa-solid fa-warehouse"></i>
                        </div>
                        <div className="card-body">
                            <StockStatusChecker/>
                        </div>
                    </div>
                </div>
            </div>
            {/* SCHEDULE AND STOCK STATUS */
            }
        </>
    )
        ;
}

export default Summary;