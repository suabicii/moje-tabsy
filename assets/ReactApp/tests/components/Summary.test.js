import React from "react";
import ReactShallowRenderer from "react-test-renderer/shallow";
import Summary from "../../components/Summary";

it('should correctly render summary', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Summary/>);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
});