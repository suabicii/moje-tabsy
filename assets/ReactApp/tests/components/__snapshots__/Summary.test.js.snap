// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render summary 1`] = `
<React.Fragment>
  <h1
    className="text-center"
  >
    Podsumowanie
  </h1>
  <DrugList
    isEditMode={false}
  />
  <div
    className="row mt-3"
  >
    <div
      className="col-md-6"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Dzisiaj muszÄ™ zaÅ¼yÄ‡ 
          <i
            className="fa-solid fa-clock"
          />
        </div>
        <div
          className="card-body"
        >
          <ul
            className="list-unstyled"
          >
            <li>
              <span
                className="fw-bold"
              >
                18:00
              </span>
               â€” Xanax: 1 szt.
            </li>
            <li>
              <span
                className="fw-bold"
              >
                20:00
              </span>
               â€” Witamina C: 1 szt.
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="col-md-6 mt-3 mt-lg-0"
    >
      <div
        className="card"
      >
        <div
          className="card-header"
        >
          Stan zapasÃ³w 
          <i
            className="fa-solid fa-warehouse"
          />
        </div>
        <div
          className="card-body"
        >
          <h5
            className="card-title text-uppercase text-bg-danger text-center rounded"
          >
            Uwaga! ðŸ˜±
          </h5>
          <p
            className="card-text"
          >
            Zapasy nastÄ™pujÄ…cych lekÃ³w/suplementÃ³w ulegajÄ… wyczerpaniu:
          </p>
          <ul
            className="list-unstyled"
          >
            <li
              className="text-danger"
            >
              Metanabol 
              <span
                className="fw-bold"
              >
                4/10 amp. 10 ml
              </span>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>
</React.Fragment>
`;
